orderbook/src/lib.rs

```diff
@@ -13,30 +13,26 @@ use anyhow::{anyhow, Context as _, Result};
 use contracts::GPv2Settlement;
 use database::trades::TradeRetrieving;
 use fee::EthAwareMinFeeCalculator;
+use futures::Future;
 use model::DomainSeparator;
-use shared::{
-    metrics::{serve_metrics, DEFAULT_METRICS_PORT},
-    price_estimation::PriceEstimating,
-};
+use shared::price_estimation::PriceEstimating;
 use std::{net::SocketAddr, sync::Arc};
 use tokio::{task, task::JoinHandle};
 
-pub fn serve_task(
+pub fn serve_api<T>(
     database: Arc<dyn TradeRetrieving>,
     orderbook: Arc<Orderbook>,
     fee_calculator: Arc<EthAwareMinFeeCalculator>,
     price_estimator: Arc<dyn PriceEstimating>,
     address: SocketAddr,
+    shutdown_receiver: impl Future<Output = T> + Send + 'static,
```

Comment:
Should be hardcoded to Output=() imo because the T is never used.

Reply:
The issue is that oneshot::channel receiver returns a Result instead of () https://docs.rs/tokio/1.12.0/tokio/sync/oneshot/struct.Receiver.html#impl-Future. I guess I can somehow map that to ()

Reply:
You should be able to do the same trick you did below:
```rust
serve_api(...,  async { shutdown_receiver.await; });
```